packages/blade/src/components/Toast/_decisions/decisions.md

```diff
@@ -0,0 +1,251 @@
+# Toast 🏷️
+
+Toasts are used to provide feedback to the user after an important action has been performed. 
+Toasts can also be used to provide feedback to the user when a system event occurs, such as when a file is saved.
+
+<img width="426" alt="image" src="./toast-thumbnail.png" />
+
+## Design
+
+- [Toast - Figma Design](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?type=design&node-id=7665-27414&mode=design&t=UNInCMmP1iFCu9je-0)
+
+## Features
+
+- Stackable
+- Auto dismissable / manual dismissable
+- Positioning
+- `informational` or `promotional` types
+
+## Anatomy
+
+<img src="./toast-anatomy.png" width="100%" alt="Toast anatomy" />
+
+## Simple Usage
+
+```jsx
+import { BladeProvider, ToastContainer, useToast } from "@razorpay/blade/components"
+
+const HomePage = () => {
+  const { showToast } = useToast();
+
+  return (
+    <Button
+      onClick={() => {
+        showToast({
+          type: 'informational',
+          color: 'success',
+          content: 'Payment Successful',
+          leading: <DollarIcon />,
+          autoDismiss: true,
+          onDismissButtonClick: () => {
+            console.log('Toast dismissed');
+          },
+          action: {
+            text: 'View',
+            onClick: () => {
+              console.log('Toast action clicked');
+            }
+          }
+        });
+      }}
+    >
+      Show Toast
+    </Button>
+  );
+};
+
+const App = () => {
+  return (
+    <BladeProvider>
+      <ToastContainer position="bottom-left" />
+      <HomePage />
+    </BladeProvider>
+  )
+}
+```
+
+## Props
+
+### ToastContainer
+
+```ts
+type ToastContainer = {
+  position: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
+}
+```
+
+```ts
+type Toast = {
+  /**
+   * @default `informational`
+   */
+  type: 'informational' | 'promotional';
+
+  /**
+   * @default `neutral`
+   */
+  color: 'neutral' | 'positive' | 'negative' | 'warning' | 'information'
+
+  /**
+   * If the type is `promotional`, the content will be `React.ReactNode`
+   */
+  content: string | React.ReactNode;
+
+  /**
+   * Can be used to render an icon
+   */
+  leading?: IconComponent;
```

Comment:
since we don't have trailing counterpart and we know that icon will always be leading wouldn't it be better to name it just `icon`? so it will also be consistent with rest of the components?

Reply:
+1. Lets just call it `icon` similar to how we do it for button

Reply:
In promotional toast the leading can also become an asset, so `leading` maybe more flexible naming here. 

<img width="478" alt="image" src="https://github.com/razorpay/blade/assets/35374649/9f6b9138-b13e-45fd-8994-0557cb7bcd66">

Reply:
What do we mean by asset? It can be an image or text or video? Any restrictions or is it a slot?

Reply:
No restrictions, it's a slot. we should atleast keep the prop name generic to cater to future usecase. 

Same thing we did for BottomSheetHeader too.

Reply:
maybe we can have icon and asset as a naming then?

Reply:
Two props? Will cause impossible states. 

<Toast icon={InfoIcon} asset={CircleIcon} />